extension/ui/components/input_bar/InputBarContainer.tsx

```diff
@@ -56,22 +58,32 @@ export const InputBarContainer = ({
 }: InputBarContainerProps) => {
   const platform = usePlatform();
   const suggestions = usePublicAssistantSuggestions(agentConfigurations);
-
-  const [nodeCandidate, setNodeCandidate] = useState<string | null>(null);
+  const [nodeOrUrlCandidate, setNodeOrUrlCandidate] = useState<
+    UrlCandidate | NodeCandidate | null
+  >(null);
   const [selectedNode, setSelectedNode] =
     useState<DataSourceViewContentNodeType | null>(null);
 
-  const handleUrlDetected = useCallback((nodeId: string | null) => {
-    if (nodeId) {
-      setNodeCandidate(nodeId);
-    }
-  }, []);
+  const handleUrlDetected = useCallback(
+    (candidate: UrlCandidate | NodeCandidate | null) => {
+      if (candidate) {
+        setNodeOrUrlCandidate(candidate);
+      }
+    },
+    []
+  );
+
+  const isAttachedFromDataSourceActivated: boolean = true;
```

Comment:
Ok so this is the one hiding the pasting and the search 👍🏻

Reply:
Indeed!